backend/authBackend.js

```
@@ -1,65 +1,35 @@
-//
-// THE HMAC CANNOT BE LEAKED!!!!!!!!!!!!!!
-const {
-  REACT_APP_OAUTH_CLIENT_ID,
-  JWT_HMAC_KEY,
-  REACT_APP_VALID_HOSTED_DOMAIN,
-} = process.env;
-
-const client = new OAuth2Client(REACT_APP_OAUTH_CLIENT_ID);
-
-/**
- * Makes the necessary requests to Google auth servers
- * to verify that the client user is actually who they
- * say they are.
- *
- * @param { string } token - idToken returned by clientside auth
- */
-async function verifyGoogleUser(token) {
-  const ticket = await client.verifyIdToken({
-    idToken: token,
-    audience: REACT_APP_OAUTH_CLIENT_ID,
-  });
-  return ticket.getPayload();
+//throw error once user is not a valid domain
+export async function throwErr() {
+  throw new Error(`User not valid, ${REACT_APP_VALID_HOSTED_DOMAIN} is correct domain.`)
 }
 
+// validates if email is the right REACT_APP_VALID_HOSTED_DOMAIN
 /**
- * Authenticates a Google user whose token was returned
- * from the client-side Google authentication flow. Assuming
- * the user is valid, a side-wide JWT will be returned for
- * future identification.
  *
- * @param { string } googleToken - googleToken returned by clientside auth
- *
- * @return { string } - site-wide JWT that allows future access
+ * @return {boolean} - returns if user is REACT_APP_VALID_HOSTED_DOMAIN matches their email
  */
-/* @expose */
-export async function authenticateUser(googleToken) {
-  const payload = await verifyGoogleUser(googleToken);
-  const { hd } = payload;
-  if (hd === REACT_APP_VALID_HOSTED_DOMAIN) {
-    // store some basic (unreliable) profile information in the token
-    return jwt.sign({
-      name: payload.name,
-      email: payload.email,
-    }, JWT_HMAC_KEY);
-  } else {
-    // TODO: Format error to meet specification defined in contentBackend
-    throw new Error(`User cannot be authenticated, ${hd} is not a valid host domain`);
+/** @expose */
+export async function validateUser() {
+  const profile = getCurrentUser(false);
```

Pass `true` here and it will throw an error instead of returning `undefined`.

---

Again, why not throw at the source?